torchmetrics/text/rouge.py

```diff
@@ -0,0 +1,144 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from torch import Tensor
+
+from torchmetrics import Metric
+from torchmetrics.functional.text.rouge import _rouge_score_compute, _rouge_score_update
+from torchmetrics.utilities.imports import _NLTK_AVAILABLE, _ROUGE_SCORE_AVAILABLE
+
+if _ROUGE_SCORE_AVAILABLE:
+    from rouge_score.rouge_scorer import RougeScorer
+    from rouge_score.scoring import BootstrapAggregator
+else:
+    RougeScorer, BootstrapAggregator = object, object
+
+
+class ROUGEScore(Metric):
+    """
+    Calculate `ROUGE score <https://en.wikipedia.org/wiki/ROUGE_(metric)>`_, used for automatic summarization.
+
+    Args:
+        preds:
+            An iterable of predicted sentences.
+        targets:
+            An iterable of target sentences.
+        newline_sep:
+            New line separate the inputs.
+        use_stemmer:
+            Use Porter stemmer to strip word suffixes to improve matching.
+        rouge_keys:
+            A list of rouge types to calculate.
+        decimal_places:
+            The number of digits to round the computed the values to.
+        compute_on_step:
+            Forward only calls ``update()`` and returns None if this is set to False. default: True
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When `None`, DDP
+            will be used to perform the allgather.
+
+    Example:
+
+        >>> targets = "Is your name John".split()
+        >>> preds = "My name is John".split()
+        >>> rouge = ROUGEScore()   # doctest: +SKIP
+        >>> from pprint import pprint
+        >>> pprint(rouge(preds, targets))  # doctest: +NORMALIZE_WHITESPACE +SKIP
+        {'rouge1_fmeasure': 0.25,
+         'rouge1_precision': 0.25,
+         'rouge1_recall': 0.25,
+         'rouge2_fmeasure': 0.0,
+         'rouge2_precision': 0.0,
+         'rouge2_recall': 0.0,
+         'rougeL_fmeasure': 0.25,
+         'rougeL_precision': 0.25,
+         'rougeL_recall': 0.25,
+         'rougeLsum_fmeasure': 0.25,
+         'rougeLsum_precision': 0.25,
+         'rougeLsum_recall': 0.25}
+
+    References:
+        [1] ROUGE: A Package for Automatic Evaluation of Summaries by Chin-Yew Lin https://aclanthology.org/W04-1013/
+    """
+
+    def __init__(
+        self,
+        newline_sep: bool = False,
+        use_stemmer: bool = False,
+        rouge_keys: Tuple[str] = ("rouge1", "rouge2", "rougeL", "rougeLsum"),  # type: ignore
```

Comment:
should we allow single str?
```suggestion
        rouge_keys: Union[str, Tuple[str]] = ("rouge1", "rouge2", "rougeL", "rougeLsum"),  # type: ignore
```

Reply:
If a single `str` is passed I convert it into a list in the function, but we can force the user to pass in a list as the tests directly use lists.

Reply:
if we already allow single str (because we just convert) then it should just be clear from the typing :]